### DISCUSSION TITLE
Cannot build React Native iOS due to code signing issue

### DISCUSSION THREAD

**Discussion by anniewey** [codemagic-ci-cd/codemagic-docs#1406] [Q&A] [answered]
Hi, I've just started on setting up the workflow for `ios` but having issue on code signing. I've already enabled Apple Developer Portal integration with my workflow as below. Also, checked out Apple Developer portal and found out we don't have any provisioning profile created. But I'm able to build and upload to TestFlight from my local device. While I'm getting error on CM. 

Build ID: 63ec7bd6fee8de34866c9dcf

Error on CM:
```
❌  /Users/builder/clone/ios/Myapp.xcodeproj: error: "MyappStaging" requires a provisioning profile with the Associated Domains and Push Notifications features. Select a provisioning profile in the Signing & Capabilities editor. (in target 'MyappStaging' from project 'Myapp')
```

```
workflows:
  ios-staging:
    name: Build iOS version (staging)
    integrations:
      app_store_connect: APPSTORE_CONNECT_API
    environment:
      groups:
        - staging_secret
        - ios_secret
      vars:
        BUNDLE_ID: "co.myapp.staging"
        XCODE_WORKSPACE: "Myapp.xcworkspace"
        XCODE_SCHEME: "MyappStaging"
      node: v16.11.1
    scripts:
      - name: Install yarn dependencies
        script: |
          yarn install
      - name: Install CocoaPods dependencies
        script: |
          cd "$CM_BUILD_DIR/ios"
          pod install
      - name: Generate config files
        script: |
          echo $AWS_EXPORTS | base64 --decode > "$CM_BUILD_DIR/aws-exports.js"
      - name: Set up code signing settings on Xcode project
        script: |
          xcode-project use-profiles
      - name: Build IPA for distribution
        script: | 
          xcode-project build-ipa \
            --workspace "$CM_BUILD_DIR/ios/$XCODE_WORKSPACE" \
            --scheme "$XCODE_SCHEME"
    artifacts:
      - build/ios/ipa/*.ipa
      - /tmp/xcodebuild_logs/*.log
      - $HOME/Library/Developer/Xcode/DerivedData/**/Build/**/*.app
      - $HOME/Library/Developer/Xcode/DerivedData/**/Build/**/*.dSYM

**Comment by icarusdust** [comment, marked as answer]
Hi @anniewey, you are missing the key step in your setup which is code signing. Check [this sample project's code signing step](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/react-native/react-native-demo-project/codemagic.yaml#L57).

**Reply by anniewey** [reply]
I've tried to add that step before but failed directly
`No matching profiles found for bundle identifier "co.myapp.staging" and distribution type "app_store"`

**Reply by icarusdust** [reply]
It means that your provisioning profile that has `co.myapp.staging` as bundle id is not `app_store` type. If it is ad_hoc, then you must specify it as ad_hoc in yaml.

**Reply by anniewey** [reply]
But my Apple Developer portal don't have any provisioning profile. Same error if use `ad_hoc`.

**Reply by icarusdust** [reply]
You will have to create a provisioning profile manually in Apple Developer account and provide it to Codemagic.

**Reply by anniewey** [reply]
Got it, thanks! Works after I create provisioning profile in Apple Developer account
